lib/response-types/code-response-type.js

```diff
@@ -4,33 +4,189 @@
+    })
+    .tap(function() {
+      state = this.getState(request);
+    })
+    .then(function(code) {
+      return { redirectUri: this.buildSuccessRedirectUri(uri, { code: code.authorizationCode, state }), result: code };
+    })
+    .catch(function(e) {
+      if (!(e instanceof OAuthError)) {
+        e = new ServerError(e);
+      }
+
+      if (state) {
+        return { redirectUri: this.buildErrorRedirectUri(uri, { error: e, state }), result: e };
+      } else {
+        return { redirectUri: this.buildErrorRedirectUri(uri, { error: e }), result: e };
+      }
+    });
+
+};
+
+/**
+ * Generate authorization code.
+ */
+
+CodeResponseType.prototype.generateAuthorizationCode = function() {
+  if (this.model.generateAuthorizationCode) {
+    return Promise.try(this.model.generateAuthorizationCode, [], this.model);
+  }
+
+  return tokenUtil.generateRandomToken();
+};
+
+/**
+ * Get authorization code lifetime.
+ */
+
+CodeResponseType.prototype.getAuthorizationCodeLifetime = function() {
+  var expires = new Date();
+
+  expires.setSeconds(expires.getSeconds() + this.authorizationCodeLifetime);
+
+  return expires;
+};
+
+/**
+ * Get scope from the request.
+ */
+
+CodeResponseType.prototype.getScope = function(request) {
+  var scope = request.body.scope || request.query.scope;
+
+  if (!is.nqschar(scope)) {
+    throw new InvalidScopeError('Invalid parameter: `scope`');
+  }
+
+  return scope;
+};
+
+/**
+ * Get state from the request.
+ */
+
+CodeResponseType.prototype.getState = function(request) {
+  var state = request.body.state || request.query.state;
+
+  if (!this.allowEmptyState && !state) {
+    throw new InvalidRequestError('Missing parameter: `state`');
+  }
+
+  if (!is.vschar(state)) {
+    throw new InvalidRequestError('Invalid parameter: `state`');
+  }
+
+  return state;
+};
+
+/**
+ * Get redirect URI.
+ */
+
+CodeResponseType.prototype.getRedirectUri = function(request, client) {
+  return request.body.redirect_uri || request.query.redirect_uri || client.redirectUris[0];
+};
+
+/**
+ * Save authorization code.
+ */
+
+CodeResponseType.prototype.saveAuthorizationCode = function(authorizationCode, expiresAt, scope, client, redirectUri, user) {
+  var code = {
+    authorizationCode: authorizationCode,
+    expiresAt: expiresAt,
+    redirectUri: redirectUri,
+    scope: scope
+  };
+
+  return Promise.try(this.model.saveAuthorizationCode, [code, client, user], this.model);
+};
+
+/**
+ * Build a successful response that redirects the user-agent to the client-provided url.
  */
 
-CodeResponseType.prototype.buildRedirectUri = function(redirectUri) {
+CodeResponseType.prototype.buildSuccessRedirectUri = function(redirectUri, query) {
   if (!redirectUri) {
     throw new InvalidArgumentError('Missing parameter: `redirectUri`');
   }
 
   var uri = url.parse(redirectUri, true);
 
-  uri.query.code = this.code;
+  uri.query = {
+    ...uri.query,
```

Comment:
This just hit me now, object destructuring is also not available in older versions :laughing:

Reply:
Oops. 😅Fixed. 👍